src/reducers/timelines.js

```diff
@@ -1,49 +1,35 @@
 import { ADD_TOOTS } from '../constants/toots'
 
-const initialState = {
-	home : {
-		// isLoading ?
-		items : []
-	},
-	local : {
-		// isLoading ?
-		items : []
-	},
-	federated : {
-		// isLoading ?
-		items : []
-	}
-}
-
-function normalizeToots(timeline, state, toots) {
-	const currentToots = state[timeline].items
+const initialState = {}
 
-	return {
-		items : toots.map(toot => toot.id).concat(currentToots)
-	}
+function normalizeToots(newtoots, oldToots) {
+	// TODO / FIXME this is ugly.
+	return [...new Set(newtoots.map(toot => toot.id).concat(oldToots))]
```

Comment:
I'd rather be explicit.
This works but it's not easy to tell exactly what is going on here and why it is used like that.